feat: New file destination plugin (#6096)
cloudquery/cloudquery: website/pages/docs/plugins/destinations/file/overview.md

## Example

This example configures a CSV destination, to create CSV files in  `./cq_csv_output`. Note that the CSV plugin only supports `append` write-mode.

The (top level) spec section is described in the [Destination Spec Reference](/docs/reference/destination-spec).

```yaml
kind: destination
spec:
  name: "file"
  path: "cloudquery/file"
  version: "VERSION_DESTINATION_FILE"
  write_mode: "append" # file only supports 'append' mode

  spec:
    directory: "./cq_csv_output"
    format: "csv"
```

## File Spec

This is the (nested) spec used by the CSV destination Plugin.

- `directory` (string) (required)

  Directory where all CSV files will be written. A CSV file will be created per table.

- `format` (string) (required)

  Format of the output file. `json` and `csv` are supported.

- `no_rotate` (bool) (optional)

>   if set to true, the plugin will append to the same file for each table.
>   Otherwise, for every batch a new file will be created with `.uuid` suffix.

Reviewer: ```suggestion
  If set to true, the plugin will write to one file per table.
  Otherwise, for every batch a new file will be created with a different `.<UUID>` suffix.
```
